src/components/ControlPanel/ControlPanel.tsx

```diff
@@ -1,33 +1,41 @@
-import type { RefObject } from 'react';
-import { type FC } from 'react';
+import type { FC, MouseEventHandler, RefObject } from 'react';
 
 import { SelectCustom } from '@components/ui/SelectCustom/SelectCustom';
 import IconSearch from '@icons/search.svg?react';
-import type { FilterByCategory, SortOptionProps } from '@interfaces/ControlPanel';
-import type { Product, ProductCategoryName } from '@interfaces/Product';
-import { getUniqueProductCategoryNames } from '@services/product.service';
+import type { FilterByCategoryHandler, SortOptionProps } from '@interfaces/ControlPanel';
+import type { ProductFilterByCategory } from '@interfaces/Product';
 
 import styles from './ControlPanel.module.css';
 
-interface ControlPanelProps extends SortOptionProps, FilterByCategory {
-    products: Product[] | null;
+export type ProductCategoryName = Pick<ProductFilterByCategory, 'name'>;
+
+interface ControlPanelProps extends SortOptionProps {
+    filtersOptionsByCategory: ProductFilterByCategory[];
+    onFilterByCategoryClick: FilterByCategoryHandler;
+    selectedFilterByCategory: ProductFilterByCategory;
     searchInputRef: RefObject<HTMLInputElement>;
     onSearchBtnClick: () => void;
 }
 
 const ControlPanel: FC<ControlPanelProps> = ({
+    filtersOptionsByCategory,
     selectedSortOption,
-    onSortOptionChange,
-    products,
-    selectedFiltersByCategory,
+    onSortOptionClick,
+    selectedFilterByCategory,
     onFilterByCategoryClick,
     onSearchBtnClick,
     searchInputRef,
 }) => {
-    const uniqueProductCategoriesNames = getUniqueProductCategoryNames(products);
+    const getFilterButtonClassName = (filterOption: ProductFilterByCategory) =>
+        selectedFilterByCategory.id === filterOption.id ? `${styles.filterBtn} ${styles.filterBtnActive}` : styles.filterBtn;
 
-    const getFilterButtonClassName = (filterButtonName: ProductCategoryName) =>
-        selectedFiltersByCategory.includes(filterButtonName) ? `${styles.filterBtn} ${styles.filterBtnActive}` : styles.filterBtn;
+    const onCategoryClick: MouseEventHandler<HTMLButtonElement> = (event) => {
+        const clickedFilterOptionId = (event.target as HTMLButtonElement).dataset.categoryId;
+        const filterOption = filtersOptionsByCategory.find((option) => option.id.toString() === clickedFilterOptionId);
+        if (filterOption) {
+            onFilterByCategoryClick(filterOption as ProductFilterByCategory);
+        }
+    };
```

Comment:
The text in the search pannel is disappear on search submit
<img width="1017" alt="image" src="https://github.com/snischuk/fe-react-2024/assets/98031768/582c0fe9-8490-4374-b69c-bcc52f1dd40c">

Reply:
Also would be good to add searching by Enter key in this field

Reply:
fixed